rlberry/agents/agent.py

```diff
@@ -302,6 +303,51 @@ def load(cls, filename, **kwargs):
         obj.__dict__.update(tmp_dict)
         return obj
 
+    @classmethod
+    def _get_param_names(cls):
+        """Get parameter names for the Model"""
+        # fetch the constructor or the original constructor before
+        # deprecation wrapping if any
+        init = getattr(cls.__init__, "deprecated_original", cls.__init__)
+        if init is object.__init__:
+            # No explicit constructor to introspect
+            return []
+
+        # introspect the constructor arguments to find the model parameters
+        # to represent
+        init_signature = inspect.signature(init)
+        # Consider the constructor parameters excluding 'self'
+        parameters = [
+            p
+            for p in init_signature.parameters.values()
+            if p.name != "self" and p.kind != p.VAR_KEYWORD
+        ]
+
+        # Extract and sort argument names excluding 'self'
+        return sorted([p.name for p in parameters])
+
+    def get_params(self, deep=True):
+        """
+        Get parameters for this agent.
+        Parameters
```

Comment:
A very minor comment on the style of docstrings, I see in other files we have blank lines to separate different sections (Parameters, Returns, References, etc), e.g. in other files a blank line would be put on line 332. Shall we keep the same standard for all files?

Reply:
I don't really have an opinion on this. I searched in numpy docstring and pep rules but I only found this in https://www.pydocstyle.org/en/2.1.1/error_codes.html . 
We may want to check this in the check if we really want to enforce this, but I think that as long as the new header is detected by sphinx in the doc, I  am happy with or without the blank lines.

Reply:
We could use this : https://pypi.org/project/flake8-docstrings/ to integrate this into flake8 format check.